theseus/geometry/so3.py

```diff
@@ -274,3 +277,57 @@ def _copy_impl(self, new_name: Optional[str] = None) -> "SO3":
     # only added to avoid casting downstream
     def copy(self, new_name: Optional[str] = None) -> "SO3":
         return cast(SO3, super().copy(new_name=new_name))
+
+    def rotate(
+        self,
+        point: Union[Point3, torch.Tensor],
+        jacobians: Optional[List[torch.Tensor]] = None,
+    ) -> Point3:
+        self._rotate_shape_check(point)
+        batch_size = max(self.shape[0], point.shape[0])
+        if isinstance(point, torch.Tensor):
+            p = point.view(-1, 3, 1)
+        else:
+            p = point.data.view(-1, 3, 1)
```

Comment:
Do you need a check here to make sure `point` is of the correct size if passed as a `torch.Tensor`?

Reply:
> Do you need a check here to make sure `point` is of the correct size if passed as a `torch.Tensor`? Same for unrotate.

I don't think so. `Union[Point3, torch.Tensor]` has already guaranteed `point` is either `Point3` or `torch.Tensor`.

Reply:
I think @mhmukadam meant that when the tensor case is hit (line 289), check that `point.shape` is consistent with what's expected. This is already done in line 286 with the call to `self._rotate_shape_check(point)`.

BTW, note that typing hints are not enforced by Python, it's mostly a help for developers. If the user is not using `mypy`, they won't see any issue with passing arbitrary objects here, until the code crashes somewhere.